haystack/nodes/document_classifier/base.py

```diff
@@ -17,20 +17,36 @@ class BaseDocumentClassifier(BaseComponent):
     query_count = 0
     query_time = 0
 
+    def __init__(
+        self,
+        classification_field: str = None,
+        convert_to_dicts: bool = False
+    ):
+        self.classification_field = classification_field
+        self.convert_to_dicts = convert_to_dicts
+
     @abstractmethod
     def predict(self, documents: List[Document]):
         pass
 
-    def run(self, query: str, documents: List[Document]): # type: ignore
+    def run(self, documents: Union[List[dict], List[Document]]): # type: ignore
         self.query_count += 1
+        field_map = {}
+        if self.classification_field is not None:
+            field_map = {self.classification_field: "content"}
```

Comment:
If I get this right, you want to allow the classification based on custom fields by over-loading the content in `Document.content`? How will this work when users supply a list of Documents rather than Dicts? 
Wouldn't it be easier to understand if we rather adjust `prediction()` method to pick the texts from `classification_field` (see comment below)?

Reply:
Yes, you're right. I didn't realize that we get any additional field potentially for classification in the meta dict anyway. It's much cleaner to access it from there.

Reply:
done